doc/docusaurus/docs/working-with-scripts/other-optimization-techniques.md

```diff
@@ -128,6 +128,37 @@ Traces can be expensive especially in terms of script sizes.
 It is advisable to use traces during development, but to remove them when deploying your scripts on mainnet.
 Traces can be removed via the `remove-trace` plugin flag.
 
+## Using `BuiltinArray` for index-based lookups
+
+For multiple lookups by index, converting a list to a `BuiltinArray` can be more cost-effective than performing lookups on the list directly. The initial cost of converting the list to an array can be offset by the significantly faster lookups.
+
+As a rule of thumb, if you only need to perform a single lookup, it is likely not worth the conversion cost. The benefits of using a `BuiltinArray` become apparent when you need to perform several lookups on the same data structure.
+
+For instance, a single lookup at index 99 of a 100-element list demonstrates this trade-off:
+*   **Lookup Cost:** The CPU cost for the lookup on a standard Plinth list (a sum-of-products type) is 206 times higher than on a `BuiltinArray`.
+*   **Creation Cost:** However, creating the `BuiltinArray` (by first creating a list and then converting it) is 11 times more expensive than creating the list alone.
```

Comment:
Really? Creating a list is linear, creating an array is also linear, how can it be 11x worse?

Are you possibly trying to say: if you _already_ have a list, then performing a single lookup is 11x faster than converting the entire list to an array, and then performing a single lookup?

Reply:
> Really? Creating a list is linear, creating an array is also linear, how can it be 11x worse?

You're right - that comparison doesn't make sense. The benchmark was comparing referencing a compile-time list constant against doing a runtime conversion, which isn't really apples-to-apples.

I've updated the docs to remove the misleading 11x claim and focus on what actually matters: the **206x lookup performance difference**. That's the real story here.

The key points now are:
  - Arrays are much faster for lookups (206x)
  - Array creation currently requires list conversion (with overhead)
  - Future versions will add arrays to `ScriptContext` to avoid conversion costs

Thanks for catching this - the original wording was confusing.